### PR TITLE
Fix rspec

### LINKED ISSUE #605: rspec issues
right now, running the rspec tests fails quiet a lot. I tried to fix some issues we introduced due to new commands in is_docker_mac but right now i am sutck

```
 2) native_osx strategy start creates a log file
     Failure/Error: FileUtils.cp_r([host_app_template, config_file_template], env.dir)
     
     NoMethodError:
       undefined method `dir' for #<Rspec::Bash::StubbedEnv:0x00007ff2a64869f8>
     # ./spec/integration/native-osx_strategy_spec.rb:24:in `block (3 levels) in <top (required)>'
```

seems like `create_stubbed_env` should create a proper stub on that using `Rspec::Bash` .. but somehow the expected method is missing?

@michaelbaudino could you assist a little? :)

### PR DESCRIPTION
**Pull request by ignatiusreza**
close #605

### CODE REVIEW

**Comment by EugenMayer** [comment]
wow, did you just.. thank you so much.

I will review it ASAP an release beta2 and the we should follow up with 0.5.11 I guess.

Thank you for the contribution once again!

File: lib/docker-sync/dependencies/fswatch.rb [hunk 1]
```diff
@@ -8,9 +8,12 @@ def self.available?
...
       def self.ensure!
-        raise 'Fswatch cannot be installed on other platforms then MacOS' unless Environment.mac?
-        PackageManager.install_package('fswatch') unless available?
-      end
+        return if available?
+
+        PackageManager.install_package('fswatch')
+        puts "please restart docker sync so the installation of fswatch takes effect"
+        exit(1)
+    end
```
[the remark this answers is not in the document]
Comment (ignatiusreza) [reply lib/docker-sync/dependencies/fswatch.rb]: the logic doesn't really change, it moved from `lib/docker-sync/dependencies.rb` to here for better encapsulation..

- before:

   ```
   if config.fswatch_required?
     unless Fswatch.available?	
       Fswatch.ensure! # will call PackageManager.install_package('fswatch')
       puts "please restart docker sync so the installation of fswatch takes effect"	
       raise(UNSUPPORTED_OPERATING_SYSTEM)	
     end	
   end
   ```

- after:

   ```
   return if available?

   PackageManager.install_package('fswatch')
   puts "please restart docker sync so the installation of fswatch takes effect"
   exit(1)
   ```

also `available?` internally have the same guard against running fswatch in non mac environment at https://github.com/EugenMayer/docker-sync/blob/796bbfde6a920267727ce452051b0ae762386fc8/lib/docker-sync/dependencies/fswatch.rb#L4-L8

[the remark this answers is not in the document]
Comment (ignatiusreza) [reply lib/docker-sync/dependencies/fswatch.rb]: in term of running `rsync` in linux.. I thought that we haven't support that per https://github.com/EugenMayer/docker-sync/wiki/docker-sync-on-Linux#using-rsync-strategy ?

File: spec/lib/docker-sync/dependencies/fswatch_spec.rb [hunk 2]
```diff
@@ -1,6 +1,11 @@
 require 'spec_helper'
 
 RSpec.describe DockerSync::Dependencies::Fswatch do
+  before do
+    allow(DockerSync::Environment).to receive(:mac?).and_return(true)
+    allow(described_class).to receive(:exit)
+  end
+
```
[the remark this answers is not in the document]
Comment (ignatiusreza) [reply spec/lib/docker-sync/dependencies/fswatch_spec.rb]: `fswatch` is never expected to run under linux, so i'm not sure what to test there.. inside our dependencies checker, we call `Fswatch.ensure!` only when `Environment.mac?` is true https://github.com/EugenMayer/docker-sync/blob/796bbfde6a920267727ce452051b0ae762386fc8/lib/docker-sync/dependencies.rb#L13-L19

should we add another check inside `.ensure_all_for_linux!` to `raise unsupportive` if the project is configured for `fswatch`? wdyt?

File: lib/docker-sync/dependencies/fswatch.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (EugenMayer) [reply lib/docker-sync/dependencies/fswatch.rb]: We do not, that does not mean people wont try it :) It was more like a "sane error message" lesser then "adding support".

Running anything on Linux then native volumes makes no sense beside probably uid/gid mapping.

But i understand, i think that is enough for now!

File: spec/lib/docker-sync/dependencies/fswatch_spec.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (EugenMayer) [reply spec/lib/docker-sync/dependencies/fswatch_spec.rb]: I would do exactly that, so raise an unsupportive - good idea!

[the remark this answers is not in the document]
Comment (ignatiusreza) [reply spec/lib/docker-sync/dependencies/fswatch_spec.rb]: updated to forbid running fswatch on non-MacOS environments! :+1:

[the remark this answers is not in the document]
Comment (EugenMayer) [reply spec/lib/docker-sync/dependencies/fswatch_spec.rb]: Perfect, ty

**Comment by EugenMayer** [comment]
Thank you very much - once again :)

**Pull request merged**